README.md

```diff
@@ -11,10 +11,36 @@ for details on the library's purpose and functionality.
 Installing
 ----------
 
-To install, first make sure you have [Go](https://golang.org)
-version 1.3 or newer installed, then run:
+First make sure you have [Go](https://golang.org)
+version 1.3 or newer installed.
+
+Install C library dependencies.
+
+OpenSSL: [OpenSSL](https://www.openssl.org/) 
+
+Pairing-Based Cryptographay: [PBC](http://crypto.stanford.edu/pbc/)
```

Comment:
The best way I can think of to optionally build the packages is to add a build flag (http://golang.org/pkg/go/build/). Then you can do conditional compilation by building with a tag.




You would add:




// +build pbc




To the files you want to ignore. Then when you want to build with pbc you would run  "go build -tags ‘pbc’"

On Wed, Jan 7, 2015 at 1:05 PM, Bryan Ford <notifications@github.com>
wrote:

>> @@ -11,10 +11,36 @@ for details on the library's purpose and functionality.
>>  Installing
>>  ----------
>>  
>> -To install, first make sure you have [Go](https://urldefense.proofpoint.com/v2/url?u=https-3A__golang.org&d=AwICaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=kz410MAwwUfvoTjTZZulJ8o8jAUM37Js5cl5izUJywY&m=7NffofZqKS5RjShNK1-RA2o4pEehm_AGwAJKDldVL_U&s=MCYCPMcaTHwyAZN9Uz0CrGOLrRt5-u8M3rB9B8ic-PU&e= )
>> -version 1.3 or newer installed, then run:
>> +First make sure you have [Go](https://urldefense.proofpoint.com/v2/url?u=https-3A__golang.org&d=AwICaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=kz410MAwwUfvoTjTZZulJ8o8jAUM37Js5cl5izUJywY&m=7NffofZqKS5RjShNK1-RA2o4pEehm_AGwAJKDldVL_U&s=MCYCPMcaTHwyAZN9Uz0CrGOLrRt5-u8M3rB9B8ic-PU&e= )
>> +version 1.3 or newer installed.
>> +
>> +Install C library dependencies.
>> +
>> +OpenSSL: [OpenSSL](https://urldefense.proofpoint.com/v2/url?u=https-3A__www.openssl.org_&d=AwICaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=kz410MAwwUfvoTjTZZulJ8o8jAUM37Js5cl5izUJywY&m=7NffofZqKS5RjShNK1-RA2o4pEehm_AGwAJKDldVL_U&s=7CSgX7ozb2LF60Ul208-huw7GRpYBcxVQ8-WwldCv0c&e= ) 
>> +
>> +Pairing-Based Cryptographay: [PBC](https://urldefense.proofpoint.com/v2/url?u=http-3A__crypto.stanford.edu_pbc_&d=AwICaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=kz410MAwwUfvoTjTZZulJ8o8jAUM37Js5cl5izUJywY&m=7NffofZqKS5RjShNK1-RA2o4pEehm_AGwAJKDldVL_U&s=SbsGQF--G1Ou7opHepUoMGdnhpQn6JLcbEdgDF5LBwg&e= )
> Also, PBC is actually optional and not actually needed by anything in the crypto library we're currently using, so maybe we should add a note to that effect.  (All that will happen is that directory will fail to build/test.)  We actually need to figure out the "right" way to deal with optional dependencies like this in the Go environment; let me know if you run across any relevant wisdom.
> ---
> Reply to this email directly or view it on GitHub:
> https://urldefense.proofpoint.com/v2/url?u=https-3A__github.com_DeDiS_crypto_pull_3_files-23r22615733&d=AwICaQ&c=-dg2m7zWuuDZ0MUcV7Sdqw&r=kz410MAwwUfvoTjTZZulJ8o8jAUM37Js5cl5izUJywY&m=7NffofZqKS5RjShNK1-RA2o4pEehm_AGwAJKDldVL_U&s=5c_jwUjVHA-eGR27sqitOPkPNK7PGmyKRxaeBpewNdI&e=